src/triage/risklist/__init__.py

```diff
@@ -0,0 +1,171 @@
+    experiment_config = results[0]['config']
+    label_config = experiment_config['label_config']
+    original_matrix_uuid = results[0]['matrix_uuid']
+    matrix_metadata = json.loads(results[0]['matrix_metadata'])
+    feature_names = matrix_metadata['feature_names']
+    feature_config = experiment_config['feature_aggregations']
+    cohort_config = experiment_config['cohort_config']
+    timechop_config = experiment_config['temporal_config']
+    feature_start_time = timechop_config['feature_start_time']
+
+    model_info = {}
+    model_info['cohort_config'] = cohort_config
+    model_info['feature_config'] = feature_config
+    model_info['feature_names'] = feature_names
+    model_info['feature_start_time'] = feature_start_time
+    model_info['original_matrix_uuid'] = original_matrix_uuid
+    model_info['label_config'] = label_config
+
+    return model_info
+
+
+def generate_risk_list(db_engine, matrix_storage_engine, model_storage_engine, model_id, as_of_date):
+    """Generate the risk list based model_id and as_of_date
+
+    Args:
+            db_engine (sqlalchemy.db.engine)
+            model_id (int) The id of a given model in the database
+            matrix_storage_engine (catwalk.storage.matrix_storage_engine)
+            model_storage_engine (catwalk.storage.model_storage_engine)
+            as_of_date (string) a date string like "YYYY-MM-DD"
+    """
+    upgrade_db(db_engine=db_engine)
+    # 1. Get feature and cohort config from database
+    model_info = get_required_info_from_config(db_engine, model_id)
+
+    # 2. Generate cohort
+    cohort_table_name = f"production.cohort_{model_info['cohort_config']['name']}"
+    cohort_table_generator = EntityDateTableGenerator(
+        db_engine=db_engine,
+        query=model_info['cohort_config']['query'],
+        entity_date_table_name=cohort_table_name
+    )
+    cohort_table_generator.generate_entity_date_table(as_of_dates=[dt_from_str(as_of_date)])
+
+    # 3. Generate feature aggregations
+    feature_generator = FeatureGenerator(
+        db_engine=db_engine,
+        features_schema_name="production",
+        feature_start_time=model_info['feature_start_time'],
+    )
+    collate_aggregations = feature_generator.aggregations(
+        feature_aggregation_config=model_info['feature_config'],
+        feature_dates=[as_of_date],
+        state_table=cohort_table_name
+    )
+    feature_generator.process_table_tasks(
+        feature_generator.generate_all_table_tasks(
+            collate_aggregations,
+            task_type='aggregation'
+        )
+    )
+
+    # 4. Reconstruct feature disctionary from feature_names and generate imputation
+    reconstructed_feature_dictionary = {}
+    imputation_table_tasks = OrderedDict()
+    with db_engine.begin() as conn:
+        for aggregation in collate_aggregations:
+            feature_prefix = aggregation.prefix
+            feature_group = aggregation.get_table_name(imputed=True).split('.')[1].replace('"', '')
```

Comment:
we have a helper for this, no? seems icky :smile_cat:

Reply:
This section will have to be redone once #608 is merged so I think its implementation will change. Or maybe #608 will have to be changed once this is merged!

Reply:
Revisiting this; since we are prioritizing this before 608, we probably shouldn't spend too much time revising this as it will change. We may still break things out into functions for clarity, and make intention more clear with comments.

Reply:
@tweddielin